include/deal.II/base/mpi.h

```diff
@@ -573,6 +595,76 @@ namespace Utilities
       internal::all_reduce(MPI_MIN, ArrayView<const T>(values, N),
                            mpi_communicator, ArrayView<T>(minima, N));
     }
+
+    template <typename T>
+    std::vector<T>
+    gather(const MPI_Comm &comm,
+           const T &object_to_send,
+           const unsigned int root_process)
+    {
+#ifndef DEAL_II_WITH_MPI
+      (void)comm;
+      std::vector<T> v(1, object_to_send);
+      return v;
+#else
+      const auto n_procs = dealii::Utilities::MPI::n_mpi_processes(comm);
+      const auto my_rank = dealii::Utilities::MPI::this_mpi_process(comm);
+
+      Assert(root_process >= 0 && root_process < n_procs,
```

Comment:
Since `root_process` is a `unsigned int`, it is always greater or equal than 0. Would you remove that check?